parallel-install/pkg/config/kubeconfigmanager.go

```diff
@@ -0,0 +1,107 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	temporaryFilePattern = "kubeconfig-*.yaml"
+)
+
+// kubeConfigManager manages resolving kubeconfig from two possible sources: a file path or a kubeconfig content.
+type kubeConfigManager struct {
```

Comment:
I think this type is redundant if we already have `KubeconfigSource`.

All logic here can be attached to `KubeconfigSource` without having unnecessary layers and abstractions. Keep it simple :)

Reply:
According to what Magda said https://github.com/kyma-project/hydroform/pull/257#discussion_r612183716 she does not want that struct to be explicitly set in the installation config. As a result, it is being resolved under the hood. On the other hand, I find it pretty much "ugly" to pass two parameters to every method that need kubeconfig. As a result, there is a simple struct that aggregates these two parameters, so we pass it around anywhere.

Reply:
Maybe you understood me wrong. What I am proposing, is to keep `KubeconfigPath` and delete `KubeconfigManager`.

Then move all the methods from `KubeconfigManager` to `KubeconfigPath`.

Reply:
Personally I pefer the exact opposite. The external type is simple placeholder. The internal type contains logic, and we are free to change the contract. If we expose this logic as part of public API, then ANY change to this logic would be a breaking change in our public API. We don't want that. We want to be free to change the logic without affecting the users.